controllers/authentication.ts

```diff
@@ -93,13 +93,19 @@ authRouter.post(
 
       const saltRounds: number = 10;
       const passwordHash: string = await hash(body.password, saltRounds);
+      let tempCompanyName = null;
+      let tempCategory = null;
+      if (["business", "agency"].includes(body.userType)) {
+        tempCompanyName = body.companyName;
+        tempCategory = body.category;
+      }
       let user: IUserDocument = new User({
         firstName: body.firstName,
         lastName: body.lastName,
         email: body.email,
         userType: body.userType,
-        category: body.category,
-        companyName: body.companyName,
+        category: tempCategory,
+        companyName: tempCompanyName,
```

Comment:
Tässä ja alemmassa paremetrissa saattais toimia ternary tai || operaattori. Eli:
```suggestion
        category: body.category || null,
        companyName: body.companyName || null,
```
Jos categoryn arvo on falsy, niin asetetaan arvoksi operaattorin oikeanpuoleinen arvo eli tässä tapauksessa null. Silloin päästäis eroon tuosta ylemmästä includes tarkistuksesta ja apumuuttujista.